tricounty-report.R

```diff
@@ -1,216 +1,329 @@
+report_month <- max(month_counts[month_counts$year == report_year, ]$month) - 1
 
 # create_public_report() creates a public report for the given month
 create_public_report <- function(month_num) {
-  current_month_counts <- monthly_counts[monthly_counts$year == current_report_year & monthly_counts$month == month_num,c("disease","counts")]
-  current_month_counts <- subset(current_month_counts, current_month_counts$disease %in% monthly_5yr_avgs$disease)
+  
+  m_counts <- with(month_counts, 
+                   month_counts[year == report_year & month == month_num,
+                                c("disease","counts")])
+  # - Only take the rows with data in the final report
+  m_counts <- subset(m_counts, disease %in% monthly_avgs$disease)
   
   # - Create the report data frame initializing the Current_Rate column to 0
-  current_report <- data.frame(
-    Disease = monthly_5yr_avgs$disease,
+  m_report <- data.frame(
+    Disease = monthly_avgs$disease,
     Current_Rate = 0, 
-    Historical_Rate = monthly_5yr_avgs[month_num + 1]
+    Historical_Rate = monthly_avgs[month_num + 1]
   )
   
-  # - Update the Current_Rate column with values from current_month_counts
-  for (i in 1:length(current_month_counts$disease)) {
-    current_report[current_report$Disease == current_month_counts$disease[i], ]$Current_Rate <- current_month_counts$counts[i]
+  # - Update the Current_Rate column with values from m_counts
+  for (i in 1:length(m_counts$disease)) {
+    d <- m_counts$disease[i]
+    m_report[m_report$Disease == d, ]$Current_Rate <- m_counts$counts[i]
   }
   
   # - Add Trends column
-  current_report$Trend <- mapply(function(x,y) {
-    if (x > y) {
-      "HIGHER"
-    } else if (x < y) {
-      "LOWER"
-    } else {
-      "SAME"
-    }
-  }, current_report$Current_Rate, current_report[[3]])
+  m_report$Trend <- mapply(function(x, y) {
+    ifelse(x > y, "↑", ifelse(x < y, "↓", "→"))
+  }, m_report$Current_Rate, m_report[[3]])
   
   # - Wait until final step to convert disease names to public-facing versions
-  current_report$Disease <- public_disease_list$Public_name
+  m_report$Disease <- diseases$Public_name
   
-  write.csv(current_report, paste0(public_reports_folder, "public_report_",   colnames(current_report)[3], current_report_year,".csv"), row.names = FALSE)
+  # - Write to CSV file
+  write_report_csv(
+    m_report, 
+    paste0("public_report_", colnames(m_report)[3], report_year, ".csv"),
+    is.public = TRUE)
 }
 
-# - Create report table for most recent month
-create_public_report(current_report_month)
-
-# - Create report table for one month prior
-create_public_report(current_report_month - 1)
-
-# - Create report table for two months prior
-create_public_report(current_report_month - 2)
-
-
-
+# - Create report table for most recent month and for 1 and 2 months prior
+create_public_report(report_month)
+create_public_report(report_month - 1)
+create_public_report(report_month - 2)
```

Comment:
See the comment on passing objects as function arguments instead of using scoping.